src/indexes.rs

```diff
@@ -72,14 +52,41 @@ impl JsonIndex {
 pub struct Index {
     pub(crate) uid: Rc<String>,
     pub(crate) client: Client,
+    pub(crate) primary_key: Option<String>,
+    pub created_at: Option<OffsetDateTime>,
+    pub updated_at: Option<OffsetDateTime>,
 }
 
 impl Index {
+    /// Internal Function to create an [Index] from `serde_json::Value` and [Client]
+    pub(crate) fn from_value(v: serde_json::Value, client: Client) -> Result<Index, Error> {
+        #[derive(Deserialize, Debug)]
+        #[allow(non_snake_case)]
+        struct PrivIndex {
+            uid: String,
+            #[serde(with = "time::serde::rfc3339::option")]
+            updatedAt: Option<OffsetDateTime>,
+            #[serde(with = "time::serde::rfc3339::option")]
+            createdAt: Option<OffsetDateTime>,
+            primaryKey: Option<String>,
+        }
+
+        let i: PrivIndex = serde_json::from_value(v).map_err(Error::ParseError)?;
```

Comment:
This looks pretty strange to me. I think your index should derive Deserialize (and maybe serialize? I’m not sure 🤔)
And then you should use it directly instead of creating this private structure that is basically an `Index`.

Reply:
I did this because I couldn't quite figure out how to deal with `client` while deserializing. Do you have any suggestions?

Reply:
Ah yes, you're right, my bad. I'm not a fan of this solution, but it's still better than doing tricks with a hollow client that implements `Default` or an `Option<Client>` just for the only time we need to deserialize it.

Let's keep it like that :+1: